spring-integration-core/src/main/java/org/springframework/integration/codec/CompositeCodec.java

```diff
@@ -43,8 +43,10 @@ public CompositeCodec(Map<Class<?>, Codec> delegates, Codec defaultCodec) {
 		this.delegates = new HashMap<Class<?>, Codec>(delegates);
 	}
 
```

Comment:
```suggestion
	/**
	 * @param defaultCodec codec for fallback
	 * @deprecated since 6.4.6 in favor of {@link #CompositeCodec(Map, Codec)} with provided delegates.
	 */
```